### PR TITLE
Feature/101 tree tab for results

### PR DESCRIPTION
**Pull request by lfogarty98**
- Refactor old TreeTable to use IntelliJ's provided TreeTable interface
- Implement displaying of (placeholder) coverage results based on package membership

### CODE REVIEW

**Review by nikomall34 — COMMENTED** [review]
I had a look at the code and I really like it. It simplifies the treetable that we had before a lot.

**Comment by lfogarty98** [comment]
@brianneoberson and I pushed changes addressing the other aspects of this ticket apart from the refactoring of the treetable. We have implemented the creation of the tree data structure given a list of coverage reports, which are currently placeholder objects in the `TreeStructureTable` class. 

Somehow the rebase to main is showing file changes for stuff we didn't actually change, sorry about that. But essentially all of the relevant changes are located in `/visualisation/treestructure`

**Review by lheimbs — COMMENTED** [review]
Guys please remove all content that is not needed for this PR.
In this state the PR is not reviewable

**Comment by brianneoberson** [comment]
> Guys please remove all content that is not needed for this PR.
> In this state the PR is not reviewable

I was able to cleanup the commit history, now only the relevant changes are shown. :)

File: pitmutationmate/src/main/kotlin/com/amos/pitmutationmate/pitmutationmate/visualization/treestructure/TreeTableModel.kt [hunk 1]
```diff
@@ -37,6 +47,40 @@ abstract class AbstractTreeTableModel(@JvmField var root: Any) : TreeTableModel
         listenerList.remove(TreeModelListener::class.java, l)
     }
 
+    override fun getColumnCount(): Int {
+        return columnNames.size
+    }
+
+    override fun setTree(tree: JTree?) {
+        println("Not yet implemented")
```
Comment (lheimbs) [inline pitmutationmate/src/main/kotlin/com/amos/pitmutationmate/pitmutationmate/visualization/treestructure/TreeTableModel.kt:55]:
```suggestion
```

File: pitmutationmate/src/main/kotlin/com/amos/pitmutationmate/pitmutationmate/visualization/treestructure/TreeTableModel.kt [hunk 2]
```diff
@@ -2,30 +2,40 @@
...
+        return (parent as DataNode).children?.size ?: 0
     }
 
     override fun isLeaf(node: Any): Boolean {
         return getChildCount(node) == 0
     }
 
-    override fun valueForPathChanged(path: TreePath, newValue: Any) {}
+    override fun valueForPathChanged(path: TreePath?, newValue: Any?) {
+        TODO("Not yet implemented")
```
Comment (lheimbs) [inline pitmutationmate/src/main/kotlin/com/amos/pitmutationmate/pitmutationmate/visualization/treestructure/TreeTableModel.kt:34]:
```suggestion
```

**Review by lheimbs — APPROVED** [review]
Looks good. Only interesting thing would be if the columns can be sized so that for example number of classes is smaller and the package/class names get more vertical space

**Pull request merged**